src/common/OutputDataSocket.cc

```diff
@@ -298,9 +298,7 @@ void OutputDataSocket::handle_connection(int fd)
 int OutputDataSocket::dump_data(int fd)
 {
   m_lock.Lock(); 
-  list<bufferlist> l;
-  l = data;
-  data.clear();
+  list<bufferlist> l = std::move(data);
```

Comment:
as @joscollin pointed out in https://github.com/ceph/ceph/pull/16525#discussion_r129041618, this leaves `data` in an unspecified state. `data` is reused below (`data.push_back(bl);`), so we need to call `data.clear();` here after the move